snuba/utils/describer.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass
+from typing import Optional, Sequence, Tuple, Union
+
+
+class DescriptionVisitor(ABC):
+    def visit_header(self, header: Optional[str]) -> None:
+        raise NotImplementedError
+
+    def visit_description(self, desc: Description) -> None:
+        raise NotImplementedError
+
+    def visit_string(self, string: str) -> None:
+        raise NotImplementedError
+
+    def visit_tuple(self, tuple: Tuple[str, str]) -> None:
```

Comment:
This one seems out of place. It looks like if was added just to support the format of the columns in `for_schema`. I think the content should solely consist of strings or other descriptions.

Reply:
visit_tuple is not used for the `for_schema`. It is for everything that is conceptually a key value pair and you may want to render it in some sort of table. 
Example:
```
    Destination: events
    Type: LEFT
```

The tuple has a specific key value pair meaning, it is design for properties that have a property name and a property value. String instead have no semantics.
I can rename this to `visit_property` if it makes more sense.

Reply:
Yeah renaming would make this more understandable.

Reply:
done